File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_set<char> hs;
        int l = 0;
        int maxLength = 0;

        for(int r = 0; r < s.size();r++){
            while(hs.count(s[r])){
                hs.erase(s[l]);
                l++;
            }

            hs.insert(s[r]);
            maxLength = max(maxLength, r-l+1);
        }

        return maxLength;
    }
};

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {

        HashSet<Character> hs = new HashSet<>();

        int l = 0;
        int maxLength = 0;

        for (int r = 0; r < s.length(); r++) {

            while (hs.contains(s.charAt(r))) {
                hs.remove(s.charAt(l));
                l++;
            }

            hs.add(s.charAt(r));

            maxLength = Math.max(maxLength, r - l + 1);
        }

        return maxLength;
    }
}